elide-core/src/main/java/com/yahoo/elide/security/FilterExpressionCheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2016, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.security;
+
+import com.yahoo.elide.core.EntityDictionary;
+import com.yahoo.elide.core.filter.Predicate;
+import com.yahoo.elide.core.filter.expression.FilterExpression;
+import com.yahoo.elide.parsers.expression.FilterExpressionCheckOkVisitor;
+import com.yahoo.elide.security.checks.InlineCheck;
+
+import java.util.Optional;
+
+/**
+ * Check for FilterExpression
+ *
+ * @param <T> Type of class
+ */
+public abstract class FilterExpressionCheck<T> extends InlineCheck<T> {
+
+    public abstract FilterExpression getFilterExpression(RequestScope requestScope);
+
+    @Override
+    public final boolean ok(T object, RequestScope requestScope, Optional<ChangeSpec> changeSpec) {
+        FilterExpression filterExpression = getFilterExpression(requestScope);
+        return filterExpression.accept(new FilterExpressionCheckOkVisitor(object, this));
```

Comment:
You can get the dictionary that this visitor needs by casting RequestScope to the core version of it.  Not ideal, but better than creating an empty dictionary below.